core/src/main/java/hudson/model/Queue.java

```diff
@@ -864,9 +822,11 @@ private WaitingItem peek() {
      * This method is primarily added to make UI threads run faster.
      *
      * @since 1.483
+     * @deprecated Use {@link #getItems()} directly. As of 1.607 the approximation is no longer needed.
      */
+    @Deprecated
     public List<Item> getApproximateItemsQuickly() {
-        return itemsView.get();
+        return Arrays.asList(getItems());
```

Comment:
How frequently do we call this operation? I'd guess it happens frequently even in UI, so I'm a bit aware of <code>Arrays.asList</code> calls for the big queues. Maybe makes sense to cache the generated list and to update it only if the snapshot changes.

It's not important since the core has been modified